api/apps/articles/routes.coffee

```diff
@@ -1,43 +1,56 @@
 _ = require 'underscore'
 { present, presentCollection } = Article = require './model'
 { setUser } = require '../users/routes'
+User = require '../users/model.coffee'
 
 # GET /api/articles
 @index = (req, res, next) ->
-  if req.query.published isnt 'true' and (not req.query.author_id? or
-     req.query.author_id isnt req.user?._id.toString()) and
-     req.user?.type isnt 'Admin'
-    return res.err 401,
-      'Must pass author_id=me to view unpublished articles. Or pass ' +
-      'published=true to only view published articles.'
+  return res.err(401, 'Must pass channel_id to view unpublished articles. Or pass ' +
+    'published=true to only view published articles.') if req.query.published is 'false' and not req.query.channel_id
+
+  access = User.hasChannelAccess req.user, req.query.channel_id
+  return res.err(401,
+    'Must be a member of this channel to view unpublished articles. ' +
+    'Pass published=true to only view published articles.') if req.query.published is 'false' and not access
+
   Article.where req.query, (err, results) ->
     return next err if err
     res.send presentCollection results
 
 # GET /api/articles/:id
 @show = (req, res, next) ->
-  if req.article.published or req.user?.type is 'Admin' or
-     req.article.author_id.equals req.user?._id
+  return res.send present req.article if req.article.published
+
+  if User.hasChannelAccess req.user, (req.article.channel_id or req.article.partner_channel_id)
     res.send present req.article
   else
     res.err 404, 'Article not found.'
 
 # POST /api/articles
 @create = (req, res, next) ->
+  unless User.hasChannelAccess req.user, (req.body.channel_id or req.body.partner_channel_id)
```

Comment:
I think it would make sense to [DRY](https://www.wikiwand.com/en/Don't_repeat_yourself) this up by moving it to the [find](https://github.com/kanaabe/positron/blob/f24ec2c3018d825b2bf4f18f68bd8a7171b50c73/api/apps/articles/routes.coffee#L66) middleware. Something after [this line](https://github.com/kanaabe/positron/blob/f24ec2c3018d825b2bf4f18f68bd8a7171b50c73/api/apps/articles/routes.coffee#L69) like....

````coffeescript
channelID = (
  req.article.channel_id or
  req.article.partner_channel_id or
  req.body.channel_id or
  req.body.partner_channel_id
)
res.err(401, 'Unauthorized) unless User.hasChannelAccess req.user, channelID
````